cron.js

```diff
@@ -0,0 +1,217 @@
+                    console.debug(`Executing job ${jobId} ...`);            
+                    
+                    // http action
+                    if (jobSpec.http) {
+                        this.httpHandler.send(tenant, jobSpec.http).then(() => {
+                            console.debug(`... Succeeded to execute job ${jobId}`);
+                        }).catch(error => {
+                            console.log(error);
+                            // TODO: generate notification
+                            console.debug(`... Failed to execute job ${jobId}`);
+                        });
+                    }
+
+                    // broker action
+                    if (jobSpec.broker) {
+                        this.brokerHandler.send(tenant, jobSpec.broker).then(() => {
+                            console.debug(`... Succeeded to execute job ${jobId}`);
+                        }).catch(error => {
+                            // TODO: generate notification
+                            console.debug(`... Failed to execute job ${jobId} (${error})`);
+                        });
+                    }
+
+                    // TODO
+                    // jscode action
+                    //if (jobSpec.jscode) {
+                    //    let script = new vm.Script(jobSpec.jscode);
+                    //    script.runInContext(this.context);
+                    //}
+                });
+
+                // cache
+                let  key = this._makeKey(tenant, jobId);
+                let value = {spec: jobSpec, job: job};
+                this.crontab.set(key, value); 
+
+                // db
+                // TODO
+
+                // start job
+                job.start();
+                
+                console.info(`Succeeded to schedule job ${JSON.stringify(jobSpec)}`);
+
+                resolve(jobId);
+            }
+            catch(ex) {
+                console.warn(`Failed to schedule job ${Json.stringify(jobSpec)} (${ex})`);
+                reject(new InternalError('Internal Error'));
+            }
+        });
+    }
+
+    readJob(tenant, jobId) {
+        return new Promise((resolve, reject) => {
+            let key = this._makeKey(tenant, jobId);
+            let value = this.crontab.get(key);
+            // found
+            if (value) {
+                resolve({jobId: jobId, spec: value.spec});
+            }
+            // not found
+            else {
+                reject(new JobNotFound(`Not found job ${jobId} for tenant ${tenant}`));
+            }
+        });
+    }
+
+    readAllJobs(tenant) {
+        return new Promise((resolve) => {
+            let jobs = [];
+            for(let [key, value] of this.crontab) {
+                let [_tenant, jobId] = key.split(':');
+                if(_tenant === tenant) {
+                    jobs.push({jobId: jobId, spec: value.spec});
+                }
+            }
+            resolve(jobs);
+        });
+    }
+
+/*
```

Comment:
some reason to this method be commented?

Reply:
Just because I haven't had time to validate it.